R/tbl-df.r

```diff
@@ -133,3 +133,10 @@ subset_rows <- function(x, i) {
     x[i]
   }
 }
+
+#' @export
+`names<-.tbl_df` <- function(x, value) {
+  names <- minimal_names(as.character(value), length(x))
```

Comment:
Just so we think through this out loud:

You think it's better to silently fix the proposed names versus checking them for minimal-ity and erroring? The latter is somewhat pedantic but also feels more like the emerging tidyverse/r-lib style.

I appreciate that we get the same outcome either way, which is that tibbles formed and managed through conventional means will have minimal names.

Reply:
I think I'm just afraid of downstream failures here, but we can try. `base::names<-()` already errs if more names are assigned than expected:

``` r
x <- 1:3
names(x) <- 1:4
#> Error in names(x) <- 1:4: 'names' attribute [4] must be the same length as the vector [3]
```

<sup>Created on 2018-10-19 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Reply:
I think the more realistic scenarios are where someone does `names(df) <- NULL` or assigns names containing `NA` or names that are too short. Should tibble silently assign different names (minimal names) or raise an error?

Reply:
I think these examples we should ponder. Is it better to error to assign names other than what was requested? The `NULL` example is the main one that gives me pause.

``` r
devtools::load_all("~/rrr/tibble")
#> Loading tibble

df <- tibble(a = 1:3, b = 4:6, c = 7:9)

names(df) <- NULL
names(df)
#> [1] "" "" ""

df <- data.frame(a = 1:3, b = 4:6, c = 7:9)

names(df) <- NULL
names(df)
#> NULL

df <- tibble(a = 1:3, b = 4:6, c = 7:9)

names(df) <- "hi"
names(df)
#> [1] "hi" ""   ""

df <- data.frame(a = 1:3, b = 4:6, c = 7:9)

names(df) <- "hi"
names(df)
#> [1] "hi" NA   NA

df <- tibble(a = 1:3, b = 4:6, c = 7:9)

names(df) <- c("hi", NA, "bye")
names(df)
#> [1] "hi"  ""    "bye"

df <- data.frame(a = 1:3, b = 4:6, c = 7:9)

names(df) <- c("hi", NA, "bye")
names(df)
#> [1] "hi"  NA    "bye"
```

<sup>Created on 2018-10-19 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>